### PR TITLE
chore(changelog): update changelog for v3.3.0

### PR DESCRIPTION
**Pull request by joshua-goldstein**
Update changelog for upcoming release.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,22 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
+## [3.3.0] - 2022-02-23
+
+This minor release fixes a bug in the maxHeaderSize parameter that could lead to panics.  We introduce an external magic number to keep track of external dependencies.  We bump up the minimum required Go version to 1.19.  No changes were made to the format of data on disk.
```
Comment (mangalaman93) [inline CHANGELOG.md:8]: limit line width

Comment (mangalaman93) [reply CHANGELOG.md:8]: fix spaces before `We introduce`

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,22 @@ All notable changes to this project will be documented in this file.
...
+
+### Fixed
+- fix: update maxHeaderSize #1800 #1877
```
Comment (mangalaman93) [inline CHANGELOG.md:11]: can we remove the first PR number, that is not needed

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,22 @@ All notable changes to this project will be documented in this file.
...
+- feat(externalMagic): Introduce external magic number (#1745) #1852
+- fix(bench): bring in benchmark fixes from main #1863 
+
+### Chores
+- upgrade go to 1.19 #1868 
+- enable linters (gosimple, govet, lll, unused, staticcheck, errcheck, ineffassign, gofmt) #1871 #1870 #1876 
+- remove dependency on io/ioutil #1879 
+- standardize import order #1869
```
Comment (mangalaman93) [inline CHANGELOG.md:19]: we probably don't need this?

**Review by mangalaman93 — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (joshua-goldstein) [reply CHANGELOG.md:8]: is line width in scope for the changelog?  The summary will not appear very good with short lines, and our previous releases have lines that wrap around.

File: CHANGELOG.md (hunk 2 above)
Comment (joshua-goldstein) [reply CHANGELOG.md:11]: done

File: CHANGELOG.md (hunk 1 above)
Comment (mangalaman93) [reply CHANGELOG.md]: I think markdown cleans up newlines while rendering. It shouldn't be a problem

**Review by akon-dey — APPROVED** [review]
LGTM! Thank you!

File: CHANGELOG.md (hunk 1 above)
Comment (joshua-goldstein) [reply CHANGELOG.md]: I tried - it doesn't, it will render on multiple lines

**Review by mangalaman93 — APPROVED** [review]

**Pull request merged**